govcd/org_test.go

```diff
@@ -174,6 +175,86 @@ func (vcd *TestVCD) Test_Admin_GetVdcByName(check *C) {
 	check.Assert(err, IsNil)
 }
 
+// Tests org function GetVDCByName with the vdc specified
+// in the config file. Then tests with a vdc that doesn't exist.
+// Fails if the config file name doesn't match with the found vdc, or
+// if the invalid vdc is found by the function.  Also tests an vdc
+// that doesn't exist. Asserts an error if the function finds it or
+// if the error is not nil.
+func (vcd *TestVCD) Test_CreateVdc(check *C) {
+	if vcd.skipAdminTests {
+		check.Skip("Configuration org != 'Sysyem'")
+	}
+
+	results, err := vcd.client.QueryWithNotEncodedParams(nil, map[string]string{
+		"type":   "providerVdc",
+		"filter": fmt.Sprintf("(name==%s)", vcd.config.VCD.ProviderVdc.Name),
+	})
+	check.Assert(err, IsNil)
+	providerVdcHref := results.Results.VMWProviderVdcRecord[0].HREF
+
+	results, err = vcd.client.QueryWithNotEncodedParams(nil, map[string]string{
+		"type":   "providerVdcStorageProfile",
+		"filter": fmt.Sprintf("(name==%s)", vcd.config.VCD.ProviderVdc.StorageProfile),
+	})
+	check.Assert(err, IsNil)
+	providerVdcStorageProfileHref := results.Results.ProviderVdcStorageProfileRecord[0].HREF
+
+	adminOrg, err := GetAdminOrgByName(vcd.client, vcd.org.Org.Name)
+	check.Assert(err, IsNil)
+	check.Assert(adminOrg, Not(Equals), AdminOrg{})
+	vdcConfiguration := &types.VdcConfiguration{
+		Name:            TestCreateOrgVdc,
+		Xmlns:           "http://www.vmware.com/vcloud/v1.5",
+		AllocationModel: "ReservationPool",
+		ComputeCapacity: []*types.ComputeCapacity{
+			&types.ComputeCapacity{
+				CPU: &types.CapacityWithUsage{
+					Units:     "MHz",
+					Allocated: 1024,
+					Limit:     1024,
+				},
+				Memory: &types.CapacityWithUsage{
+					Allocated: 1024,
+					Limit:     1024,
+				},
+			},
+		},
+		VdcStorageProfile: &types.VdcStorageProfile{
+			Enabled: true,
+			Units:   "MB",
+			Limit:   1024,
+			Default: true,
+			ProviderVdcStorageProfile: &types.Reference{
+				HREF: providerVdcStorageProfileHref,
+			},
+		},
+		ProviderVdcReference: &types.Reference{
+			HREF: providerVdcHref,
+		},
+	}
+	task, err := adminOrg.CreateVdc(vdcConfiguration)
+	check.Assert(task, Equals, Task{})
+	check.Assert(err, Not(IsNil))
+	check.Assert(err.Error(), Equals, "VdcConfiguration missing required field: ComputeCapacity[0].Memory.Units")
+	vdcConfiguration.ComputeCapacity[0].Memory.Units = "MB"
+
+	err = adminOrg.CreateVdcWait(vdcConfiguration)
+	check.Assert(err, IsNil)
+
```

Comment:
All tests, after creating an entity, need to add the entity to the cleanup list (using function `AddtoCleanupList`), and there must be a cleanup section for VDC inside `removeLeftoverEntities` (currently, there is an empty place holder for VDC). The purpose of this method is to remove leftovers after failed operations.

Reply:
Sweet thanks - I added code to the switch statement for the VDC case.